build-system/compile/compile.js

```diff
@@ -389,6 +390,12 @@ function compile(
           reject(err);
         })
         .pipe(rename(outputFilename))
+        .pipe(
+          gulpIf(
+            !argv.pseudo_names && !options.skipUnknownDepsCheck,
+            checkForUnknownDeps()
```

Comment:
Can you make `checkForUnknownDeps()` call `reject` if the check fails? Similar to line 390. It's plumbed through the rest of the compilation pipeline.

Reply:
You told me to delete it https://github.com/ampproject/amphtml/pull/24284#discussion_r321505122

Reply:
The thing I asked you to delete back then was the call to `handleCompilerError()`, which is only meant to process errors spat out by closure compiler.

What I'm suggesting right now is that you call `reject` with whatever error `checkForUnknownDeps()` spits out (assuming it fails).

Unless I'm missing something?

Reply:
I think this will work:
```js
        .pipe(
          gulpIf(
            !argv.pseudo_names && !options.skipUnknownDepsCheck,
            checkForUnknownDeps()
          )
        )
        .on('error', reject)
        .pipe(sourcemaps.write('.'))
```

Reply:
Done.

Reply:
And I went in and looked at `handleCompilerError` and noticed that all it does is print a message with the filename, so it's fine to call it here. I retract my entire series of comments!